component-samples/components-oauth/main.js

```diff
@@ -0,0 +1,43 @@
+import "./styles.css";
+
+// Import the ArcGIS Map Components
+import "@arcgis/map-components/dist/components/arcgis-map";
+import "@arcgis/map-components/dist/components/arcgis-legend";
+import "@arcgis/map-components/dist/components/arcgis-search";
+
+// Import ArcGIS core modules for OAuth authentication
+import esriConfig from "@arcgis/core/config";
+import IdentityManager from "@arcgis/core/identity/IdentityManager";
+import OAuthInfo from "@arcgis/core/identity/OAuthInfo";
+
+// Configure the portal URL and OAuth settings
+esriConfig.portalUrl = "YOUR_PORTAL_URL"; // Replace with your actual Portal URL
+
+const oAuthInfo = new OAuthInfo({
+    appId: "YOUR_APP_ID", // Replace with your actual App ID
+    portalUrl: esriConfig.portalUrl,
+    popup: true,
```

Comment:
Is there a mention of needing the callback page?

Reply:
`redirectUri` isn't a valid OAuthInfo property. See callback doc at: https://developers.arcgis.com/javascript/latest/api-reference/esri-identity-OAuthInfo.html#popup

Reply:
Changed  popup:`false` and updated README to use `popupCallbackUrl` by adding `oauth-callback.html` file to the folder.